src/Sylius/Bundle/CoreBundle/spec/Processor/CatalogPromotionStateProcessorSpec.php

```diff
@@ -0,0 +1,76 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace spec\Sylius\Bundle\CoreBundle\Processor;
+
+use PhpSpec\ObjectBehavior;
+use SM\Factory\FactoryInterface;
+use SM\StateMachine\StateMachineInterface;
+use Sylius\Bundle\CoreBundle\Checker\CatalogPromotionEligibilityCheckerInterface;
+use Sylius\Bundle\CoreBundle\Processor\CatalogPromotionStateProcessorInterface;
+use Sylius\Component\Core\Model\CatalogPromotionInterface;
+use Sylius\Component\Core\Model\ChannelPricingInterface;
+use Sylius\Component\Core\Model\ProductVariantInterface;
+use Sylius\Component\Promotion\Model\CatalogPromotionTransitions;
+
+final class CatalogPromotionStateProcessorSpec extends ObjectBehavior
```

Comment:
This class checks only two conditions, while we have around 8 different paths in the source path that we can go through. Some of them seem to be broken to me - like ending up with processing CP, for the CP that can be in-state processing, cannot be deactivated, is enabled, but outside of operating dates. We need to cover the proper flow of this class. But rather than writing all these tests straight away, I would start with a discussion with PO and simplification of process by merging `isCatalogPromotionEligibleOperatingTime` into `isCatalogPromotionEligible`